cpp/src/lists/set_operations.cu

```diff
@@ -117,10 +124,23 @@ auto reconstruct_offsets(rmm::device_uvector<size_type> const& labels,
  * @brief Extract rows from the input table based on the boolean values in the input `condition`
  * column.
  */
-auto extract_if()
+auto extract_if(table_view const& input,
+                rmm::device_uvector<bool> const& condition,
+                rmm::cuda_stream_view stream,
+                rmm::mr::device_memory_resource* mr)
 
 {
-  //
+  auto gather_map = rmm::device_uvector<size_type>(input.num_rows(), stream);
+  auto const gather_map_end =
+    thrust::copy_if(rmm::exec_policy(stream),
+                    thrust::make_counting_iterator<size_type>(0),
+                    thrust::make_counting_iterator<size_type>(input.num_rows()),
+                    condition.begin(),
+                    gather_map.begin(),
+                    thrust::identity{});
+
+  return cudf::detail::gather(
+    input, gather_map.begin(), gather_map_end, out_of_bounds_policy::DONT_CHECK, stream, mr);
```

Comment:
This looks like `cudf::detail::copy_if` that we already have: https://github.com/rapidsai/cudf/blob/10c4c33286ef7330ab423e601ca0b516bdf81a07/cpp/include/cudf/detail/copy_if.cuh#L308-L327

Reply:
Oh yeah it's the optimized version of this that computes the output size before creating `gather_map` to avoid redundant memory usage.